cacli/cmd/cancel/cancel.go

```diff
@@ -0,0 +1,49 @@
+package cancel
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/cloud-annotations/training/cacli/cmd/login"
+	"github.com/cloud-annotations/training/cacli/e"
+	"github.com/jedib0t/go-pretty/text"
+	"github.com/spf13/cobra"
+)
+
+func Run(cmd *cobra.Command, args []string) {
+	if len(args) < 1 {
+		e.Exit(errors.New("no `Model ID` provided\nUsage: cacli cancel <model_id>"))
+	}
+
+	modelID := args[0]
+
+	session := login.AssertLoggedIn()
+
+	model, err := session.GetTrainingRun(modelID)
+	if err != nil {
+		e.Exit(err)
+	}
+
+	switch model.Entity.Status.State {
+	case "completed", "error", "failed", "canceled":
+		fmt.Println("training status : ", model.Entity.Status.State)
+		e.Exit(err)
+	case "pending", "running":
+		// cancel
+		s := spinner.New(spinner.CharSets[14], 60*time.Millisecond)
+		s.Suffix = " Cancelling training run..."
```

Comment:
Canceling only has 1 `l`

Reply:
I like to wrap all network calls in a spinner and `session.GetTrainingRun(modelID)` is a network call. 
That why I would like this to moved to immediately after `login.AssertLoggedIn()`
that way it would be:
```
Checking login...
Canceling training run...
success/failed
```

vs:
```
Checking login...
<...awkward pause...>
Canceling training run...
success/failed
```

Reply:
well. if the state of the training run is already one of the 4 states that does not require us to cancel why bother?

Reply:
the cancel call takes forever to return, if there is no need to try to cancel this i would like to avoid it

Reply:
I'm saying to show the spinner when you start making network calls.
the flow should be:
1. check login
2. show "canceling..." spinner
3. get the training runs state
4. If its already canceled, failed, complete, error then stop spinner and exit the program
5. otherwise cancel the run
6. stop spinner when canceled